featurebyte/session/spark.py

```diff
@@ -350,8 +350,12 @@ async def fetch_query_stream_impl(self, cursor: Any) -> AsyncGenerator[pa.Record
         schema = pa.schema(
             {metadata[0]: self._get_pyarrow_type(metadata[1]) for metadata in cursor.description}
         )
-        while True:
-            record_batch = self._read_batch(cursor, schema)
-            yield record_batch
-            if record_batch.num_rows == 0:
-                break
+        try:
+            while True:
+                record_batch = self._read_batch(cursor, schema)
```

Comment:
Is this where the TypeError will be raised? Would it be possible reduce the scope of the try catch to this line only?

Reply:
Sure, that's possible